swap/src/bin/asb.rs

```diff
@@ -73,6 +77,21 @@ async fn main() -> Result<()> {
 
     let env_config = env::Testnet::get_config();
 
+    let uac = tor::UnauthenticatedConnection::default()
+        .with_control_port(config.tor.control_port)
+        .with_socks5_port(config.tor.socks5_port);
```

Comment:
A builder pattern is only useful if the `default` instance is useful by itself. I haven't dug into the details but it looks like we always want to configure at least the socks5 port. Hence we should design this in a way so we can't make an instance of `UnauthenticatedConnection` without passing that port in.

I am also not convinced that carrying over the terminology of "Unauthenticated" vs "Authenticated" connection from torut actually helps us here in anyway. A more general "TorClient" might be a better fit.